discovery.md

```diff
@@ -600,83 +604,233 @@ A `GET` MAY be sent to the status endpoint to determine the status of the
+specified Services are to be added to the Discovery Endpoint.
 
 The follow responses are defined by this specification:
-- `201 Created` if the new Service was added. The response MUST include
-  an HTTP `Location` Header with a value that points to the new Service.
-  The HTTP Response Body MUST include the JSON represenation of the
-  newly created Service.
-- `404 Not Found` if there is no Service with the specified `id`.
+- `201 Created` if all the specified Services were processed successfully.
+  - If there was only one Service in the incoming request, then the response
+    MUST include an HTTP `Location` Header with a value that points to the
+	correspdonding Service.
+  - If the request contained more than one Service, then the HTTP `Location`
+    Header MUST NOT appear in the response.
+  - The HTTP Response Body MUST include an array of `id` values whose order
+    and values MUST match the list of Services in the request.
 
 Other responses are allowed, but not defined by this specification.
 
-#### `PUT /services/{id}`
+The format of the HTTP Request MUST adhere to the following:
+```
+Content-Type: application/json
+
+[
+  {
+    "url": "{url}",
+    "name": "{name}",
+    ... remainder of Service attributes ...
+  }
+  ...
+]
+```
 
-This MUST add or update a Service in a Discovery Endpoint. If a Service
-already exists with the specified `id` then this MUST attempt to update that
-Service. If it does not exist, then a new Service MUST be added with that `id`.
+The format of the HTTP Response MUST adhere to the following:
+```
+Content-Type: application/json
 
-The Body of the request message MUST contain a single Service
-definition. Unlike the `POST` operation, the Service MUST contain an `id`
-attribute that matches the `{id}` in the `PUT` URL.
+[
+  "id": "{id}"
+  ...
+]
+```
 
-In the case of a new Service being created, the sender MAY choose to include
-the `epoch` attribute. It will do so only in cases where it is doing an
-"import" type of operation where preserving the previously defined value
-is needed. For "create" type of operations the sender MUST NOT include it.
+#### `POST /services?import`
+
+This MUST add, or update, the specified Services to the list of Services
+available at this Discovery Endpoint. The Body of the request message MUST
+contain an array of zero or more Service Entries. For convinience to clients,
+the presence of the `id` and `epoch` attributes are OPTIONAL. If either is
+present then those value MUST be preserved by the Discovery Endpoint and
+not modified as a result of this operation. If they are not present then the
+Discovery Endpoint MUST assign appropriate values.
+
+If the Discovery Endpoint is unable to successully add all of the Services
+in the incoming request then an error MUST be generated and none of the
+specified Services are to be added, or updated, in the Discovery Endpoint.
+
+The Services in the request message MUST be processed in the order in which
```

Comment:
please read this carefully

Reply:
while I do think this is useful, and maybe even necessary in a real-world scenarios, bulk updates are always problematic and potentially complex.